### PR TITLE
Schema for internal transactions

### PR DESCRIPTION
**Pull request by AyushyaChitransh**
Schema for internal transactions https://github.com/blockchain-etl/ethereum-etl/issues/53

### CODE REVIEW

**Comment by medvedev1088** [comment]
The internal_tx_trace_id column can have MODE: REPEATED since it's an array. Then the export job will use json output format which supports REPEATED columns.

File: schemas/gcp/internal_transactions.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        "type": "STRING",
+        "description": "Gas available for the internal transaction"
+    },
+    {
+        "name": "internal_tx_gas_used",
+        "type": "STRING",
+        "description": "Gas used by the internal transaction"
+    },
+    {
+        "name": "internal_tx_trace_id",
```
Comment (medvedev1088) [inline schemas/gcp/internal_transactions.json]: Should we call it internal_tx_trace_address for consistency with Parity https://wiki.parity.io/JSONRPC-trace-module#trace_block?

**Comment by medvedev1088** [comment]
I think we can also add internal_tx_output column which corresponds to Parity's `result.result.output` field.

**Comment by medvedev1088** [comment]
For the export job implementation I think we should add [`trace_filter`](https://wiki.parity.io/JSONRPC-trace-module#trace_filter) API to web3.py (somewhere [here](https://github.com/ethereum/web3.py/blob/master/web3/parity.py)). Then the job implementation will mostly resemble the flow in `ExportErc20TransfersJob` from `export_erc20_transfers_job.py`

**Comment by medvedev1088** [comment]
@AyushyaChitransh the Issue on Github now has $400 funding https://github.com/blockchain-etl/ethereum-etl/issues/53. Would you like to continue working on it?

**Comment by AyushyaChitransh** [comment]
@medvedev1088 I have been working on this issue using a different setup, and am facing multiple performance issues. The coding part is completed, but the migration part(using geth) has been a bigger issue.

For now, I have made some progress on the performance issues related to this. Am hoping to improve the performance by this weekend.

**Comment by medvedev1088** [comment]
@AyushyaChitransh could you elaborate on the performance issues? Also are you using geth's `debug_subscribe` method described here https://github.com/ethereum/go-ethereum/pull/15516?

**Comment by AyushyaChitransh** [comment]
While I was tracing individual transaction hashes, I was having trouble tracing internal transactions for blocks issued around DAO fork(around block numbers 2.3M). These transactions were big and trace for these blocks was slow.

Geth occasionally times out with a response of:

> ERROR[08-04|08:11:16.028] write tcp x.x.x.x-> y.y.y.y: i/o timeout

or

> Tracer timed out

Now I am going to look into subscribe functions. And see if this speeds up process.

**Comment by medvedev1088** [comment]
@AyushyaChitransh interesting. Why did you give up using Parity?

**Comment by AyushyaChitransh** [comment]
I never used parity in the first place. When I came across this issue, my setup was preconfigured to use geth. But am inclined towards testing the performance of parity.

**Comment by medvedev1088** [comment]
@AyushyaChitransh got it. Let me know if you are interested in contributing for geth implementation. I will create and ask to fund an issue similar to this one but for geth https://github.com/blockchain-etl/ethereum-etl/issues/53.

`debug_subscribe` only works via websockets or ipc as I understand.

The `Tracer timed out` exception you got for `debug_traceTransaction`?

**Comment by AyushyaChitransh** [comment]
That is correct. I encountered this error for `debug_traceTransaction`. Fix for this was to increase the timeout but that slowed down the overall process.

And no, I do not recommend geth. Geth is unable to trace internal transactions related to self_destruct opcode. While parity is much suited for it. I have explored very deeply using geth and have mentioned almost all points of failure.

So I would like to contribute for parity implementation.

**Pull request closed without merging**